Update deprecation notes with example and guide links (#414)
ballerina-platform/module-ballerina-regex: README.md

  [![GitHub Last Commit](https://img.shields.io/github/last-commit/ballerina-platform/module-ballerina-regex.svg)](https://github.com/ballerina-platform/module-ballerina-regex/commits/main)
  [![Github issues](https://img.shields.io/github/issues/ballerina-platform/ballerina-standard-library/module/regex.svg?label=Open%20Issues)](https://github.com/ballerina-platform/ballerina-standard-library/labels/module%2Fregex)

> **Deprecation Notice:** This library is deprecated and will no longer be maintained or updated. Instead, it is recommended to use the [`ballerina/lang.regexp`](https://lib.ballerina.io/ballerina/lang.regexp/latest) library for continued support and updates. For more information, see the new [regexp types example](https://ballerina.io/by-example/regexp-type), [regexp operations example](https://ballerina.io/by-example/regexp-operations), and [regexp feature guide](https://ballerina.io/learn/distinctive-language-features/advanced-general-purpose-language-features/#regular-expressions).

Reviewer: ```suggestion
**Deprecation Notice:** This library is deprecated and will no longer be maintained or updated. Instead, it is recommended to use the [`ballerina/lang.regexp`](https://lib.ballerina.io/ballerina/lang.regexp/latest) library for continued support and updates. For more information, see the new [RegExp type example](https://ballerina.io/by-example/regexp-type), [RegExp operations example](https://ballerina.io/by-example/regexp-operations), and [Regular expressions feature guide](https://ballerina.io/learn/distinctive-language-features/advanced-general-purpose-language-features/#regular-expressions).
```

Reviewer: @pcnfernando

Reviewer 2: Agree with @LakshanWeerasinghe suggestion

  [![GitHub Last Commit](https://img.shields.io/github/last-commit/ballerina-platform/module-ballerina-regex.svg)](https://github.com/ballerina-platform/module-ballerina-regex/commits/main)
  [![Github issues](https://img.shields.io/github/issues/ballerina-platform/ballerina-standard-library/module/regex.svg?label=Open%20Issues)](https://github.com/ballerina-platform/ballerina-standard-library/labels/module%2Fregex)

> **Deprecation Notice:** This library is deprecated and will no longer be maintained or updated. Instead, it is recommended to use the [ballerina/lang.regexp](https://lib.ballerina.io/ballerina/lang.regexp/latest) library for continued support and updates. You can find the new regexp types example [here](https://ballerina.io/by-example/regexp-type), the new regexp operations example [here](https://ballerina.io/by-example/regexp-operations) and the regexp feature guide [here](https://ballerina.io/learn/distinctive-language-features/advanced-general-purpose-language-features/#regular-expressions

Reviewer 3: ```suggestion
**Deprecation Notice:** This library is deprecated and will no longer be maintained or updated. Instead, it is recommended to use the [`ballerina/lang.regexp`](https://lib.ballerina.io/ballerina/lang.regexp/latest) library for continued support and updates. For more information, see the new [regexp types example](https://ballerina.io/by-example/regexp-type), [regexp operations example](https://ballerina.io/by-example/regexp-operations), and [regexp feature guide](https://ballerina.io/learn/distinctive-language-features/advanced-general-purpose-language-features/#regular-expressions).
```
We should not be adding hyperlinks with the word `here1`. Instead, the best practice is to have the page/BBE title itself as the hyperlink.

Author: Fixed in d934dae
